src/object/cli_obj.c

```diff
@@ -1493,8 +1503,9 @@ shard_task_sched(tse_task_t *task, void *arg)
 			if (rc != 0)
 				goto out;
 
-			shard_auxi->map_ver		= map_ver;
-			shard_auxi->target		= target;
+			if (!obj_auxi->req_tgts.ort_srv_disp)
```

Comment:
For retry case under server dispatch mode, we also need to set_param(), but it seems done when obj_req_fanout(), right? if possible, please give some comment. Thanks!

Reply:
> For retry case under server dispatch mode, we also need to set_param(), but it seems done when obj_req_fanout(), right? if possible, please give some comment. Thanks!

right, I wanted to unify those two places' handling, but related with current code of dc_obj_query_key() that not change to obj_req_fanout() style so plan to change them together in next patch. I'll try to refine it then. thx